src/atn/LexerActionExecutor.ts

```diff
@@ -167,12 +167,16 @@ export class LexerActionExecutor implements LexerAction {
         return this.#cachedHashCode;
     }
 
+    public get cachedHashCode(): number | undefined {
+        return this.#cachedHashCode;
+    }
+
     public equals(other: LexerActionExecutor): boolean {
         if (this === other) {
             return true;
         }
 
-        if (this.#cachedHashCode !== other.#cachedHashCode) {
+        if (this.cachedHashCode !== other.cachedHashCode) {
```

Comment:
There is also a `hashCode` method that computes & sets the hash code if undefined- maybe it's better to use that instead of creating a getter like I did, but that would be a behavior change & I'm not sure if there might be unintended consequences.